File: src/52.ts
// Computes the smallest positive integer x such that 2x, 3x, 4x, 5x, and 6x, contain the same digits.
//
// See {@link https://projecteuler.net/problem=52}
export default function compute() {
  let x = 1;

  while (true) {
    const s = sort(x);
    const isValid = [2, 3, 4, 5, 6].every(m => {
      return sort(x * m) === s;
    });

    if (isValid) {
      return x;
    }

    x++;
  }
}

function sort(x: number): string {
  return x.toString().split('').sort().join('');
}

File: project_euler/52.py
# Problem 52: Permuted Multiples
#
# Find the smallest positive integer, x, such that 2x, 3x, 4x, 5x, and 6x, contain the same digits.
#
# See https://projecteuler.net/problem=52
def sort_digits(n: int) -> str:
    return "".join(sorted(str(n)))


def run() -> int:
    n = 1
    while True:
        s = sort_digits(n)
        if all(sort_digits(n * m) == s for m in range(2, 7)):
            return n
        n += 1


def test_run():
    assert run() == 142857
